clang/lib/Basic/Sanitizers.cpp

```diff
@@ -46,26 +47,20 @@ bool clang::parseSanitizerWeightedValue(StringRef Value, bool AllowGroups,
 #include "clang/Basic/Sanitizers.def"
                                  .Default(SanitizerMask());
 
-  if (ParsedKind) {
-    size_t equalsIndex = Value.find_first_of('=');
-    if (equalsIndex != llvm::StringLiteral::npos) {
-      double arg;
-      if ((Value.size() > (equalsIndex + 1)) &&
-          !Value.substr(equalsIndex + 1).getAsDouble(arg)) {
-        // AllowGroups is already taken into account for ParsedKind,
-        // hence we unconditionally expandSanitizerGroups.
-        SanitizerMask ExpandedKind = expandSanitizerGroups(ParsedKind);
-
-        for (unsigned int i = 0; i < SanitizerKind::SO_Count; i++)
-          if (ExpandedKind & SanitizerMask::bitPosToMask(i))
-            Cutoffs[i] = arg;
-
-        return true;
-      }
-    }
-  }
-
-  return false;
+  if (!ParsedKind)
```

Comment:
please double check this patch https://github.com/llvm/llvm-project/pull/121619/commits/a21c77b247cb557cdcf0459c50fdd47aaa67caad

Reply:
Thanks, the tests pass! Should the clamping be moved to `parseNoSanitizeHotArgs`? Hypothetically there could be some future flag, `-fsanitize-something-else=undefined=42,null=56` that could reuse `parseSanitizerWeightedValue` if clamping were not applied in `parseSanitizerWeightedValue`.

Reply:
No opinion, we can move when needed